File: src/main/python/com/datacollector/FifoQueue.py
from observable import Observable


class FifoQueue(Observable):
    _queue = []

    def enqueue(self, data):
        self._queue.append(data)
        self.trigger("toSendQueue", self._queue)

    def dequeue(self):
        if self._queue.__sizeof__() > 0:
            return self._queue.pop()
        else:
            return None

File: src/main/java/com/ibrdtnapplication/FifoQueue.java
package com.ibrdtnapplication;

import java.util.concurrent.ConcurrentLinkedDeque;
import java.util.logging.Logger;

class FifoQueue<T> extends Observable<FifoQueue<T>> {
    private static final Logger log = Logger.getLogger("FifoQueue");
    private ConcurrentLinkedDeque<T> values = new ConcurrentLinkedDeque<>();

    FifoQueue() {
        this("FifoBundleQueue");
    }

    private FifoQueue(String name) {
        super(name);
    }

    synchronized void enqueue(T value) {
        log.info("notified");
        if (value == null) return;
        boolean ret = this.values.add(value);
        log.info("ret: " + ret);
        if (ret) {
            update(this);
        }
    }

    synchronized T dequeue() {
        log.info("notified");
        return this.values.poll();
    }
}
